### PR TITLE
ref(sampling): Dynamic sampling feature flag cleanup [TET-190]

### PR DESCRIPTION
**Pull request by andriisoldatenko**
This PR depricate `filters-and-sampling` feature flag 
and replace it with `organizations:server-side-sampling`

depends on https://github.com/getsentry/sentry/pull/36504

### CODE REVIEW

**Review by priscilawebdev — APPROVED** [review]
lgtm

File: src/sentry/features/__init__.py [hunk 1]
```diff
@@ -82,7 +82,7 @@
 default_manager.add("organizations:discover-events-rate-limit", OrganizationFeature, True)
 default_manager.add("organizations:duplicate-alert-rule", OrganizationFeature, True)
 default_manager.add("organizations:enterprise-perf", OrganizationFeature)
-default_manager.add("organizations:filters-and-sampling", OrganizationFeature, True)
+default_manager.add("organizations:server-side-sampling", OrganizationFeature, True)
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/features/__init__.py:85]: we have this in this line and in the line 145 https://github.com/getsentry/sentry/blob/6e2609ad10ac6a5936f766b14c17fe60338e0eb7/src/sentry/features/__init__.py#L145

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply src/sentry/features/__init__.py]: fixed in e5276e0019

**Review by untitaker — APPROVED** [review]

**Comment by markstory** [comment]
I found a few occurrences of the existing feature name in JS code too

* static/app/views/settings/project/sampling/index.tsx
* static/app/views/settings/project/navigationConfiguration.tsx
* tests/js/spec/views/settings/project/sampling/modal.spec.tsx
* tests/js/spece/views/settings/porject/sampling/utils.tsx

Do these also need to be updated?

**Comment by andriisoldatenko** [comment]
> I found a few occurrences of the existing feature name in JS code too
> 
> * static/app/views/settings/project/sampling/index.tsx
> * static/app/views/settings/project/navigationConfiguration.tsx
> * tests/js/spec/views/settings/project/sampling/modal.spec.tsx
> * tests/js/spece/views/settings/porject/sampling/utils.tsx
> 
> Do these also need to be updated?

i think @priscilawebdev done it in https://github.com/getsentry/sentry/pull/36504

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -172,7 +172,7 @@ def _get_project_config(project, full_config=True, project_keys=None):
             "projectId": project.id,  # XXX: Unused by Relay, required by Python store
         }
     allow_dynamic_sampling = features.has(
-        "organizations:filters-and-sampling",
+        "organizations:server-side-sampling",
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply src/sentry/relay/config/__init__.py:175]: we are going to add some logic to the backend, so this feature flag is not controlled only by flagr. There is a Jira task for it

[the remark this answers is not in the document]
Comment (jjbayer) [reply src/sentry/relay/config/__init__.py:175]: Perfect, thanks @priscilawebdev!

**Review by jjbayer — APPROVED** [review]

**Pull request merged**